### PR TITLE
Adding Unit Tests For The Implemented Methods

### PR DESCRIPTION
**Pull request by Odongo006**
I am submitting this pull request to seek your review and feedback on the recent additions to the codebase. This pull request adds unit tests for methods implemented previously:

## Changes Made
- ### The changes made in this pull request adhere to industry-standard coding practices, and comprehensive unit testing.
- ### Added unit tests for all the implemented methods to ensure their correctness and robustness.

I kindly request you to review the code, provide feedback, and suggest any improvements or changes you deem necessary. Your expertise and insights are highly appreciated.

### CODE REVIEW

File: music/genre.rb [hunk 1]
```diff
@@ -1,15 +1,15 @@
+require '../../item'
```
Comment (ReemMohamedAbdelfatah) [inline music/genre.rb:1]: cantfile

**Review by ReemMohamedAbdelfatah — CHANGES_REQUESTED** [review]
# STATUS : Changes Required ♻️

Hi @Odongo006   😃, 

You've really done great work, I must confess!

There are some issues that you still need to work on to go to the next project but you are almost there!

Error404GIF (2)


## Highlights
Github flow observed✔️
Linters Set up correctly ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: music/genre.rb (hunk 1 above)
Comment (ReemMohamedAbdelfatah) [reply music/genre.rb:1]: - [ ] Hello @Odongo006, It is a requirement to "add unit tests for all implemented methods" So please make sure you implement them for all methods

**Review by torobucci — APPROVED** [review]
# Hi @Odongo006  👋
You have done very well in this PR. 👍 
Congratulations! 🎉 
ApprovedCheckGIF


Nothing to say except it's time to merge it.
 
## Highlights 🌟
✅ Correct Github WorkFlow.
✅ No Linter errors.
✅ Good commit messages 
✅ Good unit tests

## Optional changes 💯 

- None

*I strongly recommend you take every comment with the [OPTIONAL] prefix into account as they can make your code better.* 

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the Issues thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

Happy coding! 💻

**Pull request merged**